python/ray/tune/trial.py

```diff
@@ -224,18 +224,33 @@ def __init__(self,
         if trainable_cls:
             default_resources = trainable_cls.default_resource_request(
                 self.config)
-            if default_resources:
-                if resources:
-                    raise ValueError(
-                        "Resources for {} have been automatically set to {} "
-                        "by its `default_resource_request()` method. Please "
-                        "clear the `resources_per_trial` option.".format(
-                            trainable_cls, default_resources))
+
+            # If Trainable returns resources, do not allow manual overrid via
+            # `resources_per_trial` by the user.
+            if default_resources and (resources or placement_group_factory):
+                raise ValueError(
+                    "Resources for {} have been automatically set to {} "
+                    "by its `default_resource_request()` method. Please "
+                    "clear the `resources_per_trial` option.".format(
+                        trainable_cls, default_resources))
+
+            # New way: Trainable returns a PlacementGroupFactory object.
+            if isinstance(default_resources, PlacementGroupFactory):
+                placement_group_factory = default_resources
+                resources = None
+            # Set placement group factory to None for backwards compatibility.
+            else:
+                placement_group_factory = None
```

Comment:
```suggestion

            # If Trainable returns resources, do not allow manual override via
            # `resources_per_trial` by the user.
            if default_resources:
                if resources or placement_group_factory:
                    raise ValueError(
                        "Resources for {} have been automatically set to {} "
                        "by its `default_resource_request()` method. Please "
                        "clear the `resources_per_trial` option.".format(
                            trainable_cls, default_resources))

                # New way: Trainable returns a PlacementGroupFactory object.
                if isinstance(default_resources, PlacementGroupFactory):
                    placement_group_factory = default_resources
                    resources = None
                # Set placement group factory to None for backwards compatibility.
                else:
                    placement_group_factory = None
                    resources = default_resources
```
(and remove line under this).

We need to keep the indent here, and with the suggested change we keep the same logic as before.

Reply:
I think the indent is not necessary here:
Even if default_resources is None, then both `placement_group_factory` and `resources` will be None, then `resources` will be set to `Resources(cpu=1)` below, and all should be good.
This will keep it simpler.

Reply:
done